tools/myst/host/host.c

```diff
@@ -369,3 +369,27 @@ int main(int argc, const char* argv[], const char* envp[])
         return 1;
     }
 }
+
+long myst_tcall_readline(const char* prompt, char* buf, size_t count)
+{
+    char* lineptr = NULL;
+    size_t n;
+
+    if (!prompt || !buf)
+        return -EINVAL;
+
+    printf("%s", prompt);
+
+    if (getline(&lineptr, &n, stdin) < 0)
+        return -errno;
+
+    if (!lineptr)
+        return -ENOMEM;
+
+    return (long)snprintf(buf, count, "%s", lineptr);
```

Comment:
Should we free `lineptr` after dumping to buf?

Reply:
I rewrote the function and moved it inside the kernel (and remove MYST_TCALL_READLINE)